.github/proverCiScripts/benchmarks_result_reporting/reporting_modules.py

```diff
@@ -0,0 +1,191 @@
+            'cpu_all_Max'           : sysstat['cpu_all_Max'],
+            'cpu_count'             : sysstat['cpu_count'],
+            'sysstats_url'          : f'{env["grafana_dashboard_prefix"]}{test_id}',
+            'logsurl'               : f'{env["s3endpoint"]}{test_id}.tar.gz'
+        }
+
+    except Exception as e:
+        print(e)
+
+    test_id = r['test_id']
+    r = pd.DataFrame([r])
+    r = r.set_index('test_date')
+    
+    return r
+
+def write_mem_time(engine, mem_statistics, test_id, dummy=False):
+    '''
+    adds mem stats df as time series data to table mem_stats
+    '''
+    table = 'testresults_cbmemtime'
+    mem_statistics['dummy'] = mem_statistics['timestamp'].apply(lambda x: f'{dummy}')
+    mem_statistics['test_id'] = mem_statistics['timestamp'].apply(lambda x: f'{test_id}')
+    mem_statistics.to_sql(table,engine,if_exists='append',index=False)
+
+def write_cpuall_time(engine, cpu_statistics, test_id, dummy=False):
+    '''
+    adds cpu stats df as time series data to table mem_stats
+    '''
+    table = 'testresults_cbcpualltime'
+    cpu_statistics['dummy'] = cpu_statistics['timestamp'].apply(lambda x: f'{dummy}')
+    cpu_statistics['test_id'] = cpu_statistics['timestamp'].apply(lambda x: f'{test_id}')
+    cpu_statistics.to_sql(table,engine,if_exists='append',index=False)
+
+def calc_stats(cstats,mstats): 
+    '''
+    returns 2 dataframes with cpu/mem stats to be consumed by postgresql engine
+    returns a dict with average/max cpu and max ram utilization durint the benchmark
+    '''
+    dfcpu,cpus = load_stats(cstats)
+    cpustats,cpu_all_Max,cpu_all_Average = process_cpustats(dfcpu)
+    dfmem, _ = load_stats(mstats)
+    memstats,max_ram = process_memstats(dfmem)
+    sysstat = {
+        'cpu_all_Average': cpu_all_Average,
+        'cpu_all_Max'    : cpu_all_Max,
+        'cpu_count'      : cpus,
+        'max_ram'        : f'{max_ram}Gb'
+    }
+    
+    return cpustats, memstats, sysstat
+
+def log_processor(pull_request,circuit, degree):
+    '''
+    Exports test metadata and result metrics from prover logfile
+    '''
+    logfile = [i for i in os.listdir('/home/ubuntu/') if 'proverlog' in i][0]
+    f = open(f'/home/ubuntu/{logfile}', 'r')
+    logdata = f.read()
+    logdata = logdata.split("\n")
+    running = [i for i in logdata if 'running' in i and 'test' in i][0].split()[1]
+    if running != '0':
+        r = [i.split(":")[1].split(".")[0].replace(" ","") for i in logdata if 'test result' in i][0]
+        if r == 'ok':
+            result = 'PASSED'
+            try:
+                sg = ''.join(g[0] for g in itertools.groupby([i for i in logdata if 'End' in i and 'Setup generation' in i ][0])).split('.', 1)[-1]
+            except:
+                sg = 'None'
+            try:
+                pg = ''.join(g[0] for g in itertools.groupby([i for i in logdata if 'End' in i and 'Circuit Proof generation' in i ][0])).split('.', 1)[-1]
```

Comment:
Sorry, I think I misunderstood #1207. 
You were looking for a match of "Circuit Proof generation." But before #1207, some benchmarks have messages like "Bytecode Proof generation with {} rows," which doesn't match.

So in your original proposed change of #1207, you changed  "Bytecode Proof generation with {} rows" to "Bytecode Circuit Proof generation with {} rows." But I mistook you were looking for a match of "Bytecode Circuit" and gave review feedback that adds constant `BENCHMARK_ID="Bytecode Circuit"`. After applying that change, it did not make the parsing clear.

This is the rust code post #1207

```rust
// circuit-benchmarks/src/bytecode_circuit.rs

//Unique string used by bench results module for parsing the result
const BENCHMARK_ID: &str = "Bytecode Circuit";
let proof_message = format!("{} Proof generation with degree = {}", BENCHMARK_ID, degree);
```
And now, in the Python script, we are looking for a substring match for 'Circuit Proof generation' from the output `proof_message`, which is hard to reason.

---
Here's my proposal.

1. We add common constants in bench_params for parsing. 

```rust
// bench_params.rs
// These prefixes are designed to be parsed by the benchmark server.
const SETUP_PREFIX: &str = "[Setup generation]";
const PROOF_PREFIX: &str = "[Proof generation]";
```

2. In each benchmark code, we add the constants to the messages

```rust
let proof_message = format!("{} {} with degree = {}", PROOF_PREFIX, BENCHMARK_ID, degree);
```

3. In the reporting_modules.py, we add constants too.

```python
# We search for the prefix in the logs of the Rust code.
PROOF_PREFIX = '[Proof generation]'
pg = ''.join(g[0] for g in itertools.groupby([i for i in logdata if 'End' in i and PROOF_PREFIX in i ][0])).split('.', 1)[-1]
```

We can apply all the above changes in this PR, so the parsing logic is visible in one place.

Reply:
> You were looking for a match of "Circuit Proof generation." But before #1207, some benchmarks have messages like "Bytecode Proof generation with {} rows," which doesn't match.

My initial intention was to simply make sure that we have common constant patterns in the logs, that the parser would always match. I think your first proposal served that purpose fine, but yes, this approach is cleaner and more consistent.